src/main/java/io/wispforest/lavender/structure/StructureTemplate.java

```diff
@@ -25,21 +29,43 @@
 import net.minecraft.world.biome.ColorResolver;
 import net.minecraft.world.chunk.light.LightingProvider;
 import org.apache.commons.lang3.mutable.MutableInt;
+import org.apache.commons.lang3.tuple.MutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
-import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.EnumMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Objects;
 import java.util.Optional;
 import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.stream.StreamSupport;
 
-public class StructureTemplate {
+public class StructureTemplate implements Iterable<Pair<BlockPos, BlockStatePredicate>> {
+
+    private static final char AIR_BLOCKSTATE_KEY = '_';
```

Comment:
Why are there random newlines between all of these fields and why did you pull apart the size definitions? For one, that was not required to accomplish the changes *and* it is extremely inconsistent with the rest of the codebase

Reply:
the newlines are because of my formatter's default settings (see: comment about editorconfig), and I pulled apart the size fields just due to habit.

I'll revert that as well.